server/service/hosts.go

```diff
@@ -1025,13 +1030,43 @@ func (svc *Service) RefetchHost(ctx context.Context, id uint) error {
 		}
 	}
 
+	if host != nil && (host.Platform == "ios" || host.Platform == "ipados") {
+		err := svc.verifyMDMConfiguredAndConnected(ctx, host)
+		if err != nil {
+			return err
+		}
+		err = svc.mdmAppleCommander.DeviceInformation(ctx, []string{host.UUID}, fleet.RefetchCommandUUIDPrefix+uuid.NewString())
+		if err != nil {
+			return ctxerr.Wrap(ctx, err, "refetch host with MDM")
+		}
+		return nil
+	}
+
 	if err := svc.ds.UpdateHostRefetchRequested(ctx, id, true); err != nil {
```

Comment:
Should you also hit this for iOS/iPadOS so that the button is looping until Fleet gets the MDM response?

Reply:
Yes, that makes sense. This way frontend users won't keep hitting refresh.

Reply:
@lucasmrod I updated the PR with the change